src/core/middlewares/rtkQueryError.ts

```diff
@@ -1,17 +1,36 @@
 import type { Middleware, MiddlewareAPI } from '@reduxjs/toolkit';
 import { isRejectedWithValue } from '@reduxjs/toolkit';
+import { forEach } from 'lodash';
 
 import toast from '@/components/Toast';
 import Events from '../events';
 
+function isErrorObject(target: unknown): target is { title: string; traceId: string; type: string; status: 400; errors: Record<string, string[]>; } {
```

Comment:
Explain what this black magic is and why not
`isErrorObject(target: { title: string; traceId: string; type: string; status: 400; errors: Record<string, string[]> })`

Reply:
Also, use `const function = () => {}` notation

Reply:
It's a type guard. It tells typescript to assume `target` is the given type if the function returns true.